clang/lib/CIR/CodeGen/CIRGenBuiltin.cpp

```diff
@@ -564,6 +568,29 @@ static mlir::Type getIntrinsicArgumentTypeFromAST(mlir::Type iitType,
   return iitType;
 }
 
+// Ensures a pointer argument matches the expected CIR pointer type,
+// emitting an addrspacecast for address-space mismatches only.
+static mlir::Value getCorrectedPtr(mlir::Value argValue, mlir::Type expectedTy,
+                                   CIRGenBuilderTy &builder) {
+  mlir::Type argType = argValue.getType();
```

Comment:
Because this doesn't exist in OG, let's just add a few asserts to clean this up:
```
auto ptrType = mlir::dyn_cast<cir::PointerType>(argType);
assert(ptrType && "expected pointer type");
assert(ptrType.getPointee() != expectedPtrType.getPointee() && "types should not match");
```